File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_MessageDataOne.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_MessageDataOne
*/
public class t_MessageDataOne : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public uint	m_id;
	public string	m_value;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_MessageDataOne() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_id = 0;
		m_value = "";
	}
	public bool read(CReadVariable cVariable) {
		try {
			m_id = cVariable.getU32();
			cVariable.getString(ref m_value,16384);
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(m_id);
		cVariable.put(ref m_value,16384);
		return true;
	}
};

File: Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_MessageDataOne.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
#/==========================================================================
#*!
#	@brief	t_MessageDataOne
#/
class t_MessageDataOne:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	m_id = 0	#U32
	m_value = ""	#std::string
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_id = 0
		self.m_value = ""

	def read(self,cVariable):
		self.m_id = cVariable.getU32()
		self.m_value = cVariable.getString(16384)
		return True

	def write(self,cVariable):
		cVariable.putU32(self.m_id)
		cVariable.putString(self.m_value,16384)
		return True

